test/library/postman-api-key.test.js

```diff
@@ -0,0 +1,196 @@
+                .that.is.equal('https://api.getpostman.com/collections/1234-588025f9-2497-46f7-b849-47f58b865807');
+
+            expect(requestArg.headers).to.be.an('object')
+                .that.has.property('X-Api-Key').to.equal('12345678');
+
+            expect(summary).to.be.an('object')
+                .that.has.property('collection').to.be.an('object')
+                .and.that.include({ id: 'C1', name: 'Collection' });
+
+            done();
+        });
+    });
+
+    it('should fetch environment via UID', function (done) {
+        newman.run({
+            collection: 'test/fixtures/run/single-get-request.json',
+            environment: '1234-931c1484-fd1e-4ceb-81d0-2aa102ca8b5f',
+            postmanApiKey: '12345678'
+        }, function (err, summary) {
+            expect(err).to.be.null;
+            expect(request.get.calledOnce).to.be.true;
+
+            let requestArg = request.get.firstCall.args[0];
+
+            expect(requestArg).to.be.an('object').with.keys(['url', 'json', 'headers']);
+
+            expect(requestArg.url).to.be.a('string')
+                .that.is.equal('https://api.getpostman.com/environments/1234-931c1484-fd1e-4ceb-81d0-2aa102ca8b5f');
+
+            expect(requestArg.headers).to.be.an('object')
+                .that.has.property('X-Api-Key').to.equal('12345678');
+
+            expect(summary).to.be.an('object')
+                .that.has.property('environment').to.be.an('object')
+                .and.that.include({ id: 'V1', name: 'Variable' });
+
+            done();
+        });
+    });
+
+    it('should fetch globals via UID', function (done) {
+        newman.run({
+            collection: 'test/fixtures/run/single-get-request.json',
+            globals: '1234-6863abf8-6630-4eec-b9cc-2a58f5efe589',
+            postmanApiKey: '12345678'
+        }, function (err, summary) {
+            expect(err).to.be.null;
+            expect(request.get.calledOnce).to.be.true;
+
+            let requestArg = request.get.firstCall.args[0];
+
+            expect(requestArg).to.be.an('object').with.keys(['url', 'json', 'headers']);
+
+            expect(requestArg.url).to.be.a('string')
+                .that.is.equal('https://api.getpostman.com/environments/1234-6863abf8-6630-4eec-b9cc-2a58f5efe589');
+
+            expect(requestArg.headers).to.be.an('object')
+                .that.has.property('X-Api-Key').to.equal('12345678');
+
+            expect(summary).to.be.an('object')
+                .that.has.property('globals').to.be.an('object')
+                .and.that.include({ id: 'V1', name: 'Variable' });
+
+            done();
+        });
+    });
+
+    it('should fetch all resources via UID', function (done) {
+        newman.run({
+            collection: '1234-588025f9-2497-46f7-b849-47f58b865807',
+            environment: '1234-931c1484-fd1e-4ceb-81d0-2aa102ca8b5f',
+            globals: '1234-6863abf8-6630-4eec-b9cc-2a58f5efe589',
```

Comment:
The Postman API doesn't have a globals endpoint, this test creates confusion and should be removed.